aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -230,131 +414,129 @@ For more information, see [Scope validation in the Web Host topic](xref:fundamen
+## Default service container replacement
 
-The built-in services container is meant to serve the basic needs of the framework and most consumer applications built on it. However, developers can replace the built-in container with their preferred container. The `ConfigureServices` method typically returns `void`, but if its signature is changed to return `IServiceProvider`, a different container can be configured and returned. There are many IOC containers available for .NET. In this example, the [Autofac](https://autofac.org/) package is used.
+The built-in service container is meant to serve the basic needs of the framework and most consumer apps built on it. However, developers can replace the built-in container with their preferred container. The `Startup.ConfigureServices` method typically returns `void`. If the method's signature is changed to return an [IServiceProvider](/dotnet/api/system.iserviceprovider), a different container can be configured and returned. There are many IoC containers available for .NET. In the following example, the [Autofac](https://autofac.org/) container is used:
 
-First, install the appropriate container package(s):
+1. Install the appropriate container package(s):
 
-* `Autofac`
-* `Autofac.Extensions.DependencyInjection`
+    * [Autofac](https://www.nuget.org/packages/Autofac/)
+    * [Autofac.Extensions.DependencyInjection](https://www.nuget.org/packages/Autofac.Extensions.DependencyInjection/)
 
-Next, configure the container in `ConfigureServices` and return an `IServiceProvider`:
+2. Configure the container in `Startup.ConfigureServices` and return an `IServiceProvider`:
 
-```csharp
-public IServiceProvider ConfigureServices(IServiceCollection services)
-{
-    services.AddMvc();
-    // Add other framework services
-
-    // Add Autofac
-    var containerBuilder = new ContainerBuilder();
-    containerBuilder.RegisterModule<DefaultModule>();
-    containerBuilder.Populate(services);
-    var container = containerBuilder.Build();
-    return new AutofacServiceProvider(container);
-}
-```
+    ```csharp
+    public IServiceProvider ConfigureServices(IServiceCollection services)
+    {
+        services.AddMvc();
+        // Add other framework services
+
+        // Add Autofac
+        var containerBuilder = new ContainerBuilder();
+        containerBuilder.RegisterModule<DefaultModule>();
+        containerBuilder.Populate(services);
+        var container = containerBuilder.Build();
+        return new AutofacServiceProvider(container);
+    }
+    ```
 
-> [!NOTE]
-> When using a third-party DI container, you must change `ConfigureServices` so that it returns `IServiceProvider` instead of `void`.
+    > [!NOTE]
+    > When using a third-party dependency injection container, change `Startup.ConfigureServices` so that it returns `IServiceProvider` instead of `void`.
 
-Finally, configure Autofac as normal in `DefaultModule`:
+3. Configure Autofac normally in `DefaultModule`:
```

Comment:
I'm confused by the word "normally" in this context. Is it needed?

Reply:
> I'm confused by the word "normally" in this context. Is it needed?

I think it just fell under "conversational tone." I'm removing it now.